src/worker/jobs/support/db.ts

```diff
@@ -12,9 +16,17 @@ export interface UpsertOptions {
   trx?: knex.Transaction
   where?: string
   returning?: string[]
+  redisKey?: string
 }
 
 export function upsert (options: UpsertOptions): Promise<Object> {
+  if (options.table === 'versions') {
```

Comment:
I wouldn't so it this way. I'd only have jobs mark things as invalid (E.g. delete entries), but I'd make the front-end APIs handing caching. This way the entire project doesn't need to be re-indexed to update the cache, we just mark things as invalid (or delete them, in this case).